goal_src/engine/game/projectiles.gc

```diff
@@ -5,3 +5,3358 @@
+        (set! a1-3 (the-as vector #f))
+        )
+       )
+      )
+     (sound-trans-convert (-> s5-0 parms trans) a1-3)
+     )
+    (set! (-> s5-0 parms mask) (the-as uint 32))
+    (-> s5-0 id)
+    )
+   0
+   (none)
+   )
+  )
+
+;; definition for method 28 of type projectile-yellow
+;; INFO: Return type mismatch int vs none.
+;; Used lq/sq
+(defmethod dummy-28 projectile-yellow ((obj projectile-yellow))
+  (cond
+   ((or
+     (not (handle->process (-> obj last-target)))
+     (zero?
+      (->
+       (the-as target (handle->process (-> obj last-target)))
+       control
+       root-prim
+       prim-core
+       collide-as
+       )
+      )
+     )
+    (cond
+     ((zero? (-> obj target-count))
+      (let
+       ((s5-0
+         (find-nearest-attackable
+          (-> obj parent-base)
+          409600.0
+          (the-as uint 0)
+          (the-as uint 0)
+          (-> obj base-vector)
+          (if (logtest? (-> obj options) 160)
+           546.13336
+           8192.0
+           )
+          )
+         )
+        )
+       (let
+        ((s4-0
+          (find-nearest-attackable
+           (-> obj parent-base)
+           163840.0
+           (the-as uint 1)
+           (the-as uint 0)
+           (-> obj base-vector)
+           (if (logtest? (-> obj options) 160)
+            910.2222
+            8192.0
+            )
+           )
+          )
+         (v1-10
+          (find-nearest-attackable
+           (-> obj parent-base)
+           28672.0
+           (the-as uint 1)
+           (the-as uint 0)
+           (-> obj base-vector)
+           16384.0
+           )
+          )
+         )
+        (if s4-0
+         (set! s5-0 s4-0)
+         )
+        (if v1-10
+         (set! s5-0 v1-10)
+         )
+        )
+       (set! (-> obj last-target) (process->handle s5-0))
+       (when s5-0
+        (set!
+         (-> obj target quad)
+         (-> s5-0 root-override root-prim prim-core world-sphere quad)
+         )
+        (if (= (-> s5-0 type symbol) 'mother-spider)
+         (set! (-> obj options) (logand -2 (-> obj options)))
+         )
+        )
+       )
+      )
+     (else
+      (set! (-> obj target quad) (-> obj target-base quad))
+      )
+     )
+    )
+   (else
+    (let ((a1-8 (handle->process (-> obj last-target))))
+     (set!
+      (-> obj target quad)
+      (-> (the-as target a1-8) control root-prim prim-core world-sphere quad)
+      )
+     )
+    (if
+     (and
+      (<
+       (vector-vector-xz-distance (-> obj root-override trans) (-> obj target))
+       20480.0
+       )
+      (< 24576.0 (fabs (- (-> obj target y) (-> obj root-override trans y))))
+      )
+     (set! (-> obj last-target) (the-as handle #f))
+     )
+    )
+   )
+  (+! (-> obj target-count) 1)
+  0
+  (none)
+  )
+
+;; definition for method 27 of type projectile-blue
+;; INFO: Return type mismatch int vs none.
+;; Used lq/sq
+(defmethod dummy-27 projectile-blue ((obj projectile-blue))
```

Comment:
this is interesting. This is code for some other projectile that plays the blue eco sound I guess?